SendTemplatedLoginEmailActivity/__tests__/handler.test.ts

```diff
@@ -36,18 +45,27 @@ describe("SendTemplatedLoginEmailActivity", () => {
     jest.clearAllMocks();
   });
 
-  it("should send a login email with the data", async () => {
+  it.each`
+    title                     | payload
+    ${"fallback login email"} | ${aValidPayload}
+    ${"login email"}          | ${aValidPayloadWithMagicLink}
+  `("should send a $title with the data", async ({ payload }) => {
     const handler = getSendLoginEmailActivityHandler(
       mockMailerTransporter as any,
       emailDefaults,
-      aPublicUrl,
       aHelpDeskRef,
       mockTracker
     );
 
-    const result = await handler(context as any, aValidPayload);
+    const templateFunction = jest.spyOn(
+      payload.magic_link ? mailTemplate : fallbackMailTemplate,
+      "apply"
+    );
+
+    const result = await handler(context as any, payload);
 
     expect(result.kind).toEqual("SUCCESS");
+    expect(templateFunction).toHaveBeenCalledTimes(1);
```

Comment:
I'd rather check both calls like (spying both `templateFunction` and `fallbackTemplateFunction`)

```typescript
    expect(templateFunction).toHaveBeenCalledTimes(payload.magic_link ? 1 : 0);
    expect(fallbackTemplateFunction).toHaveBeenCalledTimes(
      payload.magic_link ? 0 : 1
    );
 ```